### ISSUE TITLE
plot cartopy with different projection distorts bus circles

### ISSUE THREAD

**Issue by FabianHofmann** [PyPSA/PyPSA#66]
The problem accounts for plotting with cartopy in non-default projection and adding pie plots for buses. One the one hand they are way too big now (because off a unnecessary rescaling), on the other they are distorted.  

Example code 

```python


import pypsa, os
import cartopy.crs as ccrs
import matplotlib.pyplot as plt

n = pypsa.Network(os.path.abspath(pypsa.__path__[0] + '/../examples/ac-dc-meshed/ac-dc-data/'))
n.lopf()

gen = n.generators.assign(g = n.generators_t.p.mean()).groupby(['bus', 'carrier']).g.sum()

n.plot(bus_sizes=gen/1e3, bus_colors={'gas':'r', 'wind':'b'})
plt.show()

n.plot(bus_sizes=gen/1e3, bus_colors={'gas':'r', 'wind':'b'}, projection=ccrs.EckertI())
plt.show()

```

It is caused by the axis_transform applied to matplotlib patch collections, after plotting them. The transformation distorts the pies, which were round before and are only round according to the default projection. 

Suggestion: plot the pies without axis_transforming them. This requires, similar to  the basemap plotting, transformation of the x,y - data and keeping the rescaling factor.

**Comment by FabianHofmann** [comment]
![default_projection](https://user-images.githubusercontent.com/19226431/57396365-8fb75500-71ca-11e9-8221-91b6790e3d58.png)
![eckert_projection](https://user-images.githubusercontent.com/19226431/57396366-8fb75500-71ca-11e9-9a94-b7152f01ccb2.png)

**Comment by coroa** [comment]
> Suggestion: plot the pies without axis_transforming them. This requires similar to  the basemap plotting, transformation of the x,y - data and keeping the rescaling factor.

I don't know what the drawback's of this are.

The correct solution are probably inset axes:

https://stackoverflow.com/questions/45266955/adding-pie-chart-at-given-coordinates-to-cartopy-projection

**Comment by coroa** [comment]
Hmm, after a quick look over the code, i think i agree plotting just the pies at transformed x,y coordinates. Would you mind preparing and testing a PR?

**Comment by FabianHofmann** [comment]
no problem, if you don't mind i would do it on a different branch in the repo, fasciliates the work flow a bit

**Comment by FabianHofmann** [comment]
PR is probably better...

**Comment by coroa** [comment]
Fixed by merging #67
